core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1247,7 +1252,7 @@ class DAGScheduler(
                 }
               }
 
-              // Note: newly runnable stages will be submitted below when we submit waiting stages
+              submitWaitingChildStages(shuffleStage)
```

Comment:
@markhamstra Thank you for your review.
Definitely we can move this into else branch.
I'll modify it.